splitio/api/telemetry.py

```diff
@@ -38,11 +43,72 @@ def record_unique_keys(self, uniques):
                 body=uniques,
                 extra_headers=self._metadata
             )
+            record_telemetry(response.status_code,  get_current_epoch_time_ms() - start, HTTPExceptionsAndLatencies.TELEMETRY, self._telemetry_runtime_producer)
             if not 200 <= response.status_code < 300:
                 raise APIException(response.body, response.status_code)
         except HttpClientException as exc:
-            _LOGGER.error(
+            _LOGGER.debug(
                 'Error posting unique keys because an exception was raised by the HTTPClient'
             )
             _LOGGER.debug('Error: ', exc_info=True)
             raise APIException('Unique keys not flushed properly.') from exc
+
+    def record_init(self, configs):
+        """
+        Send init config data to the backend.
+
+        :param configs: configs
+        :type json
+        """
+        start = get_current_epoch_time_ms()
+        try:
+            response = self._client.post(
+                'telemetry',
+                '/v1/metrics/config',
+                self._apikey,
+                body=configs,
+                extra_headers=self._metadata,
+            )
+            record_telemetry(response.status_code, get_current_epoch_time_ms() - start, HTTPExceptionsAndLatencies.TELEMETRY, self._telemetry_runtime_producer)
+            if not 200 <= response.status_code < 300:
+                raise APIException(response.body, response.status_code)
+        except HttpClientException as exc:
+            _LOGGER.debug(
+                'Error posting init config because an exception was raised by the HTTPClient'
+            )
+            _LOGGER.debug('Error: ', exc_info=True)
+            raise APIException('Init config data not flushed properly.') from exc
+
+    def record_stats(self, stats):
+        """
+        Send runtime stats to the backend.
+
+        :param stats: stats
+        :type json
+        """
+        start = get_current_epoch_time_ms()
+        try:
+            response = self._client.post(
+                'telemetry',
+                '/v1/metrics/usage',
+                self._apikey,
+                body=stats,
+                extra_headers=self._metadata,
+            )
+            record_telemetry(response.status_code, get_current_epoch_time_ms() - start, HTTPExceptionsAndLatencies.TELEMETRY, self._telemetry_runtime_producer)
+            if not 200 <= response.status_code < 300:
+                raise APIException(response.body, response.status_code)
+        except HttpClientException as exc:
+            _LOGGER.debug(
+                'Error posting runtime stats because an exception was raised by the HTTPClient'
+            )
+            _LOGGER.debug('Error: ', exc_info=True)
+            raise APIException('Runtime stats not flushed properly.') from exc
+
+class LocalhostTelemetryAPI(object):  # pylint: disable=too-few-public-methods
```

Comment:
you shouldn't need to execute any method in TelemetryAPI for localhost mode, there should be something before that is NoOp

Reply:
Removed